meilisearch-http/src/index_controller/index_actor/actor.rs

```diff
@@ -175,16 +183,26 @@ impl<S: IndexStore + Sync + Send> IndexActor<S> {
         meta: Processing<UpdateMeta>,
         data: File,
     ) -> Result<UpdateResult> {
-        debug!("Processing update {}", meta.id());
-        let uuid = meta.index_uuid();
-        let update_handler = self.update_handler.clone();
-        let index = match self.store.get(*uuid).await? {
-            Some(index) => index,
-            None => self.store.create(*uuid, None).await?,
+        let uuid = meta.index_uuid().clone();
+
+        *self.processing.write().await = Some(uuid);
+
+        let result = {
+            debug!("Processing update {}", meta.id());
+            let update_handler = self.update_handler.clone();
+            let index = match self.store.get(uuid).await? {
+                Some(index) => index,
+                None => self.store.create(uuid, None).await?,
+            };
+
+            spawn_blocking(move || update_handler.handle_update(meta, data, index))
+                .await
+                .map_err(|e| IndexError::Error(e.into()))
```

Comment:
uhm, I'm not quite sure it's ok. As I understand, if it'll return early with error, our `processing` state won't change it become unsynchronized.
I introduced the closure with the goal of capturing the `Result` first, changing the `processing` status and only then returning the result.

AFAIK, Rust doesn't have such thing as `defer` which may be useful there, so it was my workaround.

Reply:
You are right, ignore my comment

Reply:
@shekhirin just make sure that you use a closure and not a block!

Reply:
oops, I meant block 😬  I think it's better there because we need to get the result into the variable instantly. Why closure?

Reply:
because you are using `?` in the block, so this is effectively short-circuiting: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=4db8eed13148eae162e917bb3db3103e

whereas this is what you want: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=77ba81a4d5e2709f19ec34993da290d9

Reply:
TIL. Thanks, I'll fix it :)